packages/core/src/tracing/cls.js

```diff
@@ -101,7 +103,32 @@ class InstanaSpan {
     if (processIdentityProvider && typeof processIdentityProvider.getFrom === 'function') {
       this.f = processIdentityProvider.getFrom();
     }
-    this.ec = 0;
+
+    // The property span.ec is defined with a getter/setter instead of being a plain property. We need to be able to
+    // prohibit overwriting span.ec from auto-tracing instrumentations after it has been set manually via the
+    // SDK (spanHandle.markAsErroneous).
+    Object.defineProperty(this, '_ec', {
+      value: 0,
+      writable: true,
+      enumerable: false
+    });
+    Object.defineProperty(this, 'ec', {
+      get() {
+        return this._ec;
+      },
+      set(value) {
+        if (!this.ecHasBeenSetManually) {
+          this._ec = value;
+        }
+      },
+      enumerable: true
+    });
+    Object.defineProperty(this, 'ecHasBeenSetManually', {
+      value: false,
+      writable: true,
+      enumerable: false
+    });
+
```

Comment:
Using three properties here (the internal value `_ec`, the enumerable value `ec` plus `ecHasBeenSetManually` feels a bit clunky. But I currently do not have any idea how to do this more elegantly.